src/main/java/seedu/clinic/model/supplier/SupplierProductsContainKeywordsPredicate.java

```diff
@@ -0,0 +1,42 @@
+package seedu.clinic.model.supplier;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.clinic.commons.util.StringUtil;
+import seedu.clinic.model.product.Product;
+
+/**
+ * Tests that any of the {@code Product} sold by {@code Supplier} matches any of the keywords given.
+ */
+public class SupplierProductsContainKeywordsPredicate implements Predicate<Supplier> {
```

Comment:
Is there any way to reduce code duplication in this case? Like defining the abstract predicate class?